extensions/parquet/base/src/main/java/io/deephaven/parquet/base/util/CachedChannelProvider.java

```diff
@@ -67,10 +88,51 @@ public SeekableByteChannel getWriteChannel(@NotNull final Path path, final boole
         final ChannelType channelType = append ? ChannelType.WriteAppend : ChannelType.Write;
         final KeyedObjectHashMap<String, PerPathPool> channelPool = channelPools.get(channelType);
         final CachedChannel result = tryGetPooledChannel(pathKey, channelPool);
-        return result == null
-                ? new CachedChannel(wrappedProvider.getWriteChannel(path, append), channelType, pathKey)
-                : result.position(append ? result.size() : 0); // The seek isn't really necessary for append; will be at
-                                                               // end no matter what.
+        if (result != null) {
+            // The seek isn't really necessary for append; will be at end no matter what.
+            return result.position(append ? result.size() : 0);
+        }
+        final SeekableByteChannel newWriteChannel = wrappedProvider.getWriteChannel(path, append);
+        channelCreatorHelper(path, newWriteChannel);
+        return new CachedChannel(newWriteChannel, channelType, pathKey);
+    }
+
+    private void channelCreatorHelper(@NotNull final Path path, @NotNull final SeekableByteChannel newChannel)
+            throws IOException {
+        // If channel creator is already marked invalid, mark the new channels invalid.
+        // Required because CachedChannelProvider cannot return a null channel, so it returns invalid channels.
+        // TODO Should we just throw an exception here?
```

Comment:
Need opinion on this behavior.

Reply:
An exception with a clear explanation is better, I think.